Use dockerd instead of docker daemon in integration-cli

vdemeester opened the pull request:
Updating `integration-cli/daemon.go` to use `dockerd` instead of `docker daemon` to start up the daemon 🐳.

<del>Maybe we should add a test for making sure `docker daemon` for the time being ?</del> done :)

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

thaJeztah commented on integration-cli/daemon.go:
```diff
@@ -23,9 +23,7 @@ import (
 // Daemon represents a Docker daemon for the testing framework.
 type Daemon struct {
-	// Defaults to "daemon"
 	// Useful to set to --daemon or -d for checking backwards compatibility
```
Looks like this comment is no longer relevant, but perhaps we should have a test for this? i.e. that `docker daemon` still works (if there's no test yet)

vdemeester replied:
maybe :stuck_out_tongue_closed_eyes:

thaJeztah commented:
LGTM

LK4D4 commented:
@vdemeester all linux CIs failed on same test

vdemeester commented:
@LK4D4 yep.. seems fishy :sweat: I'll take a look at it :wink:

LK4D4 commented:
@vdemeester How's the progress? :)

thaJeztah commented:
ping @vdemeester :innocent:

justincormack commented:
retest this please

thaJeztah commented:
@justincormack you need to `!rebuild docker#<pr-number>/<ci-job-name>` on IRC (#docker-maintainers). The "retest this please" only triggers docs CI

justincormack commented:
Aha thanks!

justincormack commented:
ok its failing on
```
23:13:49 
23:13:49 ----------------------------------------------------------------------
23:13:49 FAIL: docker_cli_authz_unix_test.go:319: DockerAuthzSuite.TestAuthZPluginAllowEventStream
23:13:49 
23:13:49 [d50864633] waiting for daemon to start
23:13:49 [d50864633] daemon started
23:13:49 docker_cli_authz_unix_test.go:368:
23:13:49     // Fail the test
23:13:49     observer.CheckEventError(c, containerID, event, matcher)
23:13:49 github.com/docker/docker/integration-cli/_test/_obj_test/events_utils.go:134:
23:13:49     ...open github.com/docker/docker/integration-cli/_test/_obj_test/events_utils.go: no such file or directory
23:13:49 ... Error: failed to observe event `create` for 98d282fe4d9e813616ecb2605ddac81490b9065105d3980e9bb8315ee523284c. Disconnection error: EOF
23:13:49 out:
23:13:49 2016-05-24T23:13:18.496410892Z image save sha256:d9551b4026f0e2950ddb557cc640871710bf88476ca938b71499305647231b82 (name=sha256:d9551b4026f0e2950ddb557cc640871710bf88476ca938b71499305647231b82)
23:13:49 
23:13:49 
23:13:49 [d50864633] exiting daemon
23:13:50 
```

Doesn't seem immediately related, perhaps worth a rebase @vdemeester ?

vdemeester commented:
Rebased, let's see :angel:

vdemeester committed:
Use dockerd instead of docker daemon in integration-cli

Updating `integration-cli/daemon.go` to use `dockerd` instead of `docker
daemon` to start up the daemon.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester commented:
Hum rebasing did not fix the issue. Hopefully, I now fixed it :sweat_smile:

thaJeztah commented:
green! re-LGTM

ping @justincormack PTAL :smile:

LK4D4 commented:
LGTM
